Assignment 2.Rmd

```diff
@@ -75,26 +75,63 @@ plot(g,layout=layout.fruchterman.reingold)
 
 #There are many ways to change the attributes of the graph to represent different characteristics of the newtork. For example, we can color the nodes according to gender.
 
-plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
+VERTEX$gender <- as.factor(VERTEX$gender)
+plot(g,layout=layout.fruchterman.reingold, vertex.color = VERTEX$gender)
 
 #We can change the thickness of the edge according to the number of times a particular teacher has sent a particular student to the principal.
 
 plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
 
-````
+```
 
 ## Part II
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
+```{r}
+stu <- EDGE %>% group_by(to) %>% summarize(sum(count))
+tea <- EDGE %>% group_by(from)%>% summarize(sum(count))
+names(stu) <- c("id", "counts")
+names(tea) <- c("id", "counts")
+VERTEX1 <- bind_rows(stu, tea)
+
+g1 <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX1)
+plot(g1,layout=layout.fruchterman.reingold, vertex.size = VERTEX1$counts)
 
```

Comment:
Graph looks great! You can size the vertices with a `log()` so that the teacher vertices do not dwarf the student vertices.